### DISCUSSION TITLE
Dual Robot Planning Example

### DISCUSSION THREAD

**Discussion by savolla** [tesseract-robotics/tesseract#929] [Q&A] [answered]
Hi, is there any example for dual `movegroup` support and plan dual robots at the same time? I want my two robots to calculate different trajectories and move to some point using these trajectories. Robots are in the same workspace and they are close to each other. so there is a risk of collision. I couldn't find any example of doing multiple robots at the same time. Once I use two separate `movegroup`s in the code and `roslaunch` it, it gives me some cryptic red errors. I can also share the error messages but they are not very helpful for locating the cause of the problem.
 
I also sent an email to both `levi.armstrong@gmail.com` and `Levi.Armstrong@srwi.org` behalf my company. We'd be very thankful if you guys reply to us. Our process could require a professional support.

**Comment by Levi-Armstrong** [comment]
I know of two approaches I have used but there may be more. In the case where both robots are working together you can create a single chain and the puzzle piece with auxiliary axes is a good example for this use case. The other option is when you have multiple robots working in overlapping workspace where you have some node keeping track of the current active trajectories and the environment has a command for adding a trajectory as a collision object.

**Reply by savolla** [reply]
the last option is what I need. Could you provide a simple code example for that one?

**Reply by Levi-Armstrong** [reply, marked as answer]
Here is the command you need to add a trajectory as a collision object to the environment.
https://github.com/tesseract-robotics/tesseract/blob/master/tesseract_environment/include/tesseract_environment/commands/add_trajectory_link_command.h
